sites/cheerpj/src/content/docs/11-guides/implementing-native-libraries.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Implementing Native Libraries
+description: Java Native Interface (JNI) with CheerpJ
+---
+
+In this guide, we’ll explore how to use **native libraries** in CheerpJ to implement Java native methods in JavaScript.
+
+Native libraries in Java are typically loaded with the `System.loadLibrary` method allowing a Java program to call functions implemented in another language. This function is part of the **Java Native Interface (JNI)**, the same interface used to work with Java native methods, which we covered in the [`Implementing Native Methods guide`].
+
+Unlike [`native methods`], native libraries are not restricted to a single Java class. They can be accessed by multiple classes and even be shared across an entire application.
+
+## Using `System.loadLibrary` for Shared Libraries
+
+[`System.loadLibrary(String libname)`] is a Java method that loads a shared library into the application, making the native methods within the library accessible to Java code. The Java runtime searches each path in `java.library.path` for the library file with the correct format for the operating system (`.so`, `.dll`, or `.dylib` on Linux, Windows, and macOS, respectively).
+
+CheerpJ adapts this concept to load JavaScript modules, using the same `System.loadLibrary` call to load `.js` files. These files are JavaScript modules that define implementations for the native methods associated with the library.
+
+## Overview
+
+In general, we can implement native libraries in CheerpJ by following these steps:
+
+1. Create a JavaScript module that implements the native methods.
+2. Create a Java class that loads the native library and declares the native methods.
+3. Load the native library in the Java class with CheerpJ.
+
+### Creating a JavaScript Module
+
+JavaScript functions that implement native methods should follow a specific naming convention - `Java_<fully-qualified-class-name>_<method-name>`. For instance, if `com.foo.Bar` has a native method called `baz`, the function will be called `Java_com_foo_Bar_baz`
```

Comment:
I would add a small explanation on what JavaScript modules are. Maybe like this?
> A JavaScript module is a file that contains code which can be exported and imported by other files for better organization and reuse. You create modules using the `export` keyword to expose classes, methods, or other resources, and you can use `export default` to make a primary, easy-to-import item from the module.

You could then also link to the official documentation like this:
For more information on JavaScript modules, check out the official [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules).

You could add a code snippet looking like this:
```
module.js
export default {
      // Java Script code
};
```
After that you can continue with this part again: JavaScript functions that will implement native Java functions need to follow a specific naming convention etc.